beets/autotag/__init__.py

```diff
@@ -95,7 +95,9 @@ def apply_metadata(album_info, mapping):
         item.title = track_info.title
 
         if config['per_disc_numbering']:
-            item.track = track_info.medium_index or track_info.index
+            item.track = track_info.medium_index
+            if item.track is None:
+                item.track = track_info.index
```

Comment:
Since this is a little tricky, an inline condition, like this:

    item.track = track_info.medium_index if track_info.medium_index is not None else track_info.index

or a comment to the effect of "we want to let the track number be zero, but if the medium index is not provided we need to fall back to the overall index" would be helpful here.